src/ExtractTileOperations.cpp

```diff
@@ -39,12 +39,46 @@ Type amx_op_type_result_type(AMXOpType op_ty) {
     }
 }
 
+int amx_op_type_size(AMXOpType op_ty) {
+    switch (op_ty) {
+    case AMXOpType::Int8:
+        return 1;
+    case AMXOpType::Bfloat16:
+        return 2;
+    default:
+        internal_error << "Unexpected";
+        return -1;
+    }
+}
+
 const auto wild_i32 = Variable::make(Int(32), "*");
 const auto wild_i32x = Variable::make(Int(32, 0), "*");
 
 Tile<1> get_1d_tile_index(const Expr &e) {
     if (const auto *r1 = e.as<Ramp>()) {
-        return {true, r1->base, {r1->stride}, {r1->lanes}};
+
+        const auto stride_var = Variable::make(Int(32), "stride");
+        const auto v1 = Variable::make(Int(32), "v1");
+        const auto v2 = Variable::make(Int(32), "v2");
+        const auto v3 = Variable::make(Int(32), "v3");
+
+        Expr patterns[] = {
```

Comment:
For completeness, should we also look for the following:
```
((stride_var * v1) + v2) * v3,
v3 * ((stride_var * v1) + wild_i32),
(v2 + (stride_var * v1)) * v3,
v3 * (v2 + (stride_var * v1)),
```
Alternatively the check that `stride` is constant could also check if `v1` is constant.

Reply:
addressed in 33408424eaca3472e043701c1f83acb14e819f28